### ISSUE TITLE
Function not defined

### ISSUE THREAD

**Issue by thrashdin** [neil-lindquist/SLIMA#12]
I add a simple function to already loaded package. 
`(defun square (x) (* x x))`

Then I compile it with `Alt+C`
`; compiling (DEFUN SQUARE ...)`

Now, when I call it `(square 4)`
`The function CL-TEST::SQUARE is undefined.`

On Emacs there is no such problem.
Weird thing is that it doesn't happen with every package, afaic, just with _some_ of them.

**Comment by Sokolov57** [comment]
It seems it just "unloads" the definitions after some time, ~ 6-7 mins was enough.

**Comment by neil-lindquist** [comment]
So the same steps will sometimes work and sometimes fail depending on how long between when the function is defined and when it's called?  That makes it seems like it's more than just the package not getting set quite right for compiling with Alt-C, which is what it started to look like when I started poking around earlier today.

**Comment by Sokolov57** [comment]
Yeah
Well, I have no idea what's going on. They updated atom which caused some other issues already, and I'm reporting them. Maybe this one is somehow related too.

**Comment by neil-lindquist** [comment]
I added a fix to the 1.1.0 release that should help with the `Alt-C` issue.  There's another issue with `Alt-k`, and possibly `Alt-C`, that's waiting on rksm/paredit.js#22, that you may or may not be running into as well.

**Comment by Sokolov57** [comment]
Good.
I tried kind of advertise the package in Lisp discord server (you know it
exists? I m just not old enough to get into the IRC stuff I guess; discord
I use on a daily basis though). But since they are playing Emacs 24/7 this
didn't cause much interest.

**Comment by neil-lindquist** [comment]
Thanks.  I've seen references to the lisp discord server, but don't otherwise use discord, so I haven't gone on there.

**Comment by Sokolov57** [comment]
okay, I got back to atom from emacs
What happens now is that while definitions doesn't just mystically disappear
`Alt-c` compilation doesn't work still. Like nothing is defined. Although everything is defined and repl is evaluating as normal. But if I press `Alt-c` It gives me tons of warnings about new stuff undefined and this
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.36.0 x64
**Electron**: 2.0.18
**OS**: Debian GNU/Linux
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Requested measurement of a line component that is not currently rendered

```
At /usr/share/atom/resources/app/static/<embedded>:11

Error: Requested measurement of a line component that is not currently rendered
    at horizontalPositionsToMeasure.forEach (/usr/share/atom/resources/app/static/<embedded>:11:105136)
    at Map.forEach (<anonymous>)
    at TextEditorComponent.measureHorizontalPositions (/usr/share/atom/resources/app/static/<embedded>:11:104999)
    at TextEditorComponent.measureContentDuringUpdateSync (/usr/share/atom/resources/app/static/<embedded>:11:70590)
    at e.readDocument (/usr/share/atom/resources/app/static/<embedded>:11:67706)
    at ViewRegistry.performDocumentUpdate (/usr/share/atom/resources/app/static/<embedded>:11:326849)
```
Here's the related issue https://github.com/atom/atom/issues/17294

**Comment by neil-lindquist** [comment]
The PR for paredit mentioned aboved was merged, so I just published a new SLIMA version that fixes an issue of code not being loaded when compiling.  That might help with the issue.  If it doesn't fix the issue, do you have an example that gives the error?  I've haven't seen this uncaught error issue before, and I'm not sure where to even start looking.

**Comment by neil-lindquist** [comment]
I think this issue's been resolved, as per our email conversation.

**Comment by Sokolov57** [comment]
Agreed, so far no problems. Thank you very much.

**Comment by mrios22** [comment]
Hello, I am using the most recent version of Slima and I still have this issue. My version of Atom is 1.39.0 x64 on OS X. I'm using the Homebrew installation of SBCL as my lisp. I followed all of the directions in the Slima documentation, including uninstalling Atom-Slime so there are no more conflicts.

When I use Alt-K or Alt-C, the REPL says that the file compiles fine, and I am able to ``(in-package ...)`` into packages. When I list the external symbols in the package, they're all there. However, it still says that all of my functions are undefined.

**Comment by neil-lindquist** [comment]
I think I found and fixed the issue.  I will get it released in the next few days; there are a couple more things I want to improve first.   You can [use the development version](https://flight-manual.atom.io/hacking-atom/sections/contributing-to-official-atom-packages/) in the mean time.

**Comment by mrios22** [comment]
Great, thanks! I'll wait until the next edition and I'll report back.
